SPECS/tpm2-tools/tpm2-tools.spec

```diff
@@ -23,23 +28,36 @@ The source repository for the TPM (Trusted Platform Module) 2 tools
 %autosetup -p1
 
 %build
-%configure \
-    --disable-static
-
-make %{?_smp_mflags}
+sed -i "/compatibility/a extern int BN_bn2binpad(const BIGNUM *a, unsigned char *to, int tolen);" lib/tpm2_openssl.c
+%configure --disable-static
+%make_build
 
 %install
-make DESTDIR=%{buildroot} install
+%make_install %{?_smp_mflags}
+
+%if 0%{?with_check}
+%check
+if [ ! -f /dev/tpm0 ];then
+   mkdir /tmp/swtpm
+   swtpm_setup --tpm-state /tmp/swtpm --tpm2
+   swtpm socket --server type=unixio,path=/tmp/swtpm/socket --ctrl type=unixio,path=/tmp/swtpm/socket.ctrl --tpmstate dir=/tmp/swtpm --flags startup-clear --tpm2 --daemon
+   export TPM2TOOLS_TCTI=swtpm:path=/tmp/swtpm/socket
+   $(find / -name tpm2_startup) -c
```

Comment:
the `tpm2_startup` (which technically isn't needed as the swtpm --flags startup-clear did the proper startup, but running it again should be ok) and `tpm2_pcrread` binaries should be in the normal $PATH, are they not being found correctly?

Reply:
no, they aren't being found on PATH

Reply:
is it better to:
* figure out why this isn't on the path
* remove the extraneous calls to tpm2_*
* leave as it is

Reply:
tpm2_startup is definitely not on the path.  i searched all the folders.

maybe this is failing because tpm2-tools doesn't have bootstrap?

Reply:
no ... this tarball has bootstrap, though the spec isn't calling it

Reply:
the files seem to be correctly set up in the rpm (if i understand what `rpm -qlp` does):

``` shell
$ rpm -qlp ../out/RPMS/x86_64/tpm2-tools-5.5-1.azl3.x86_64.rpm | grep tpm2_pcrread
/usr/bin/tpm2_pcrread
/usr/share/man/man1/tpm2_pcrread.1.gz

bfjelds@bfjelds-xubuntu:~/kubevirt/CBL-Mariner-new/toolkit$ rpm -qlp ../out/RPMS/x86_64/tpm2-tools-5.5-1.azl3.x86_64.rpm | grep tpm2_startup
/usr/bin/tpm2_startup
/usr/share/man/man1/tpm2_startup.1.gz
```

maybe i can just delete the commands if they aren't needed.

Reply:
removed them

Reply:
ah ok i see what's going on, since this is the `tpm2-tools` package itself, those programs are being provided by this actual package, so you should be able to access them using the prefix %{buildroot}/%{_bindir}/

can you try that to see if it works? Although if not, it should be fine to not call them, they aren't actually needed to run the real `make check`

Reply:
> ah ok i see what's going on, since this is the `tpm2-tools` package itself, those programs are being provided by this actual package, so you should be able to access them using the prefix %{buildroot}/%{_bindir}/
> 
> can you try that to see if it works? Although if not, it should be fine to not call them, they aren't actually needed to run the real `make check`

add some file checks ... a few specific files (tpm2, tpm2_startup, tpm2_pcrread) and then one that finds and counts all of them and compares them against 101 (which is what i found).  i'm not sure about using a magic number like 101 .. .what do you think?

Reply:
that worked :)